src/backend/distributed/utils/metadata_cache.c

```diff
@@ -329,6 +335,12 @@ CitusHasBeenLoaded(void)
 		}
 
 		extensionLoaded = extensionPresent && extensionScriptExecuted;
+
+		if (extensionLoaded)
+		{
+			/* Cache pg_dist_partition's id so we can look for it to be invalidated */
+			DistPartitionRelationId();
```

Comment:
I believe not but it feels less brittle if this is left in.

It's important that `CitusHasBeenLoaded` is called with some method which calls `DistPartitionRelationId` because `Citus...Loaded` sets extensionLoaded, which must be cleared during a `DROP EXTENSION`. If `Dist...Id` is not called, then we will never notice `pg_dist_partition` is being dropped and never reset `extensionLoaded`.

It turns out that `Dist...Id` is indirectly called by all methods which also call `CitusHasBeenLoaded`, but there's nothing which enforces that. Considering breaking this assumption will result in subtle problems much later instead of crashes, I think it's important to spend some lines to not rely on our callers doing the right thing.